ostd/src/arch/riscv/qemu.rs

```diff
@@ -0,0 +1,26 @@
+// SPDX-License-Identifier: MPL-2.0
+
+//! Providing the ability to exit QEMU and return a value as debug result.
+
+/// The exit code of QEMU.
+#[derive(Debug, Clone, Copy, PartialEq, Eq)]
+pub enum QemuExitCode {
+    /// The code that indicates a successful exit.
+    Success,
+    /// The code that indicates a failed exit.
+    Failed,
+}
+
+/// Exit QEMU with the given exit code.
+///
+/// This function assumes that the kernel is run in QEMU with the following
+/// QEMU command line arguments that specifies the ISA debug exit device:
+/// `-device isa-debug-exit,iobase=0xf4,iosize=0x04`.
```

Comment:
Do QEMU-riscv support ISA-DEBUG-EXIT device? I see the implementation does somewhat uses SBI calls to shut QEMU down. So perhaps the documentation doesn't match the implementation.